src/components/AccountModule/AccountModule.js

```diff
@@ -1,78 +1,49 @@
-import React, { useRef } from 'react'
+import React, { useRef, useState, useCallback } from 'react'
 import PropTypes from 'prop-types'
 import {
-  Button,
   EthIdenticon,
   Popover,
   GU,
   IconDown,
-  IconConnect,
   RADIUS,
   textStyle,
   useTheme,
-  IdentityBadge,
   unselectable,
+  ButtonBase,
 } from '@aragon/ui'
 import { shortenAddress } from '../../web3-utils'
 import { useAccount } from '../../account'
-
-function getNetworkName(networkId) {
-  if (networkId === 'main') return 'Mainnet'
-  if (networkId === 'rinkeby') return 'Rinkeby'
-  return networkId
-}
+import NotConnected from './NotConnected'
+import ConnectionInfo from './ConnectionInfo'
+import { useConnectionColor, useNetworkConnectionData } from './utils'
 
 function AccountModule({ compact }) {
   const { connected } = useAccount()
-  return connected ? <ConnectedMode /> : <NonConnectedMode compact={compact} />
+  return connected ? <ConnectedMode /> : <NotConnected compact={compact} />
 }
 
 AccountModule.propTypes = {
   compact: PropTypes.bool,
 }
 
-function NonConnectedMode({ compact }) {
-  const { enable } = useAccount()
-  return (
-    <div
-      css={`
-        display: flex;
-        align-items: center;
-        text-align: left;
-        padding: 0 ${(compact ? 1 : 2) * GU}px;
-      `}
-    >
-      <Button
-        size={compact ? 'small' : 'medium'}
-        icon={<IconConnect />}
-        label="Enable account"
-        onClick={enable}
-        wide
-      />
-    </div>
-  )
-}
-
-NonConnectedMode.propTypes = {
-  compact: PropTypes.bool,
-}
-
 function ConnectedMode() {
-  const { address, label, networkId } = useAccount()
+  const { address, label } = useAccount()
   const theme = useTheme()
-  // const [opened, setOpened] = useState(false)
+  const [opened, setOpened] = useState(false)
 
-  // const open = useCallback(() => {
-  //   setOpened(true)
-  // }, [])
+  const close = useCallback(() => {
+    setOpened(false)
+  }, [])
 
-  // const close = useCallback(() => {
-  //   setOpened(false)
-  // }, [])
+  const toggle = useCallback(() => {
+    setOpened(!opened)
+  }, [opened])
```

Comment:
Wondering we could just do

```
const toggle = setOpened(opened => !opened)
```

Reply:
I think you mean:

```jsx
const toggle = () => setOpened(opened => !opened)
```

And yes I agree!

Reply:
https://github.com/aragon/aragon/pull/1137/commits/b4811a8ec5a2a024784d774834b1bff924368aac